ONSdigital/dp-dataset-api
# Added extra necessary locks for instance writer handlers

### What

- Added extra locks acquire/release's in all remaining instance writer endpoints because, although they were doing atomic updates, they still need to wait if some non-atomic write operation is taking place.
- Added comments where locks are not needed (i.e. instance getter and new instance creation)

### How to review

- Make sure code changes make sense
- Make sure unit tests pass
- Make sure new comments are clear

### Who can review

anyone (probably Rhys)
